augur/tree.py

```diff
@@ -261,13 +261,27 @@ def random_string(n):
             ofile.write(tmp_line)
 
     # Check tree builder arguments for conflicts with hardcoded defaults.
-    check_conflicting_args(tree_builder_args, ("-ntmax", "-s", "-m"))
+    check_conflicting_args(tree_builder_args, (
```

Comment:
_not at all familiar with this code, ignore if I'm just utterly confused_

why are these args hard-coded in place here, instead of being part of the `DEFAULT_ARGS` data structure around L25? digging a little bit, I see this is a pattern throughout, but I find it sort of confusing that there's a set of hardcoded default arguments for each type of tree-builder, and then we're checking for "conflicts with hardcoded defaults" when running each tree-builder, using a different set of arguments.

Reply:
I _think_ hard-coded args are arguments that are set controlled through `augur tree` options and for which we want to disallow users from overwriting them. Things like: the input alignment, which augur tree processes and sets directly, there's no reason why a user should change that, we want to make this part of the underlying tree binary "private".

Default args are recommendations for tree builder programs that user may override if they want to without causing harm.

Does that make sense?

Reply:
> Does that make sense?

It does; albeit the use of "default" to mean two orthogonal things that _look_ like they're different ways of saying the same thing (`DEFAULT_ARGS` versus "hardcoded defaults") seems inherently confusing. 

Maybe "hardcoded defaults" could be changed to "hardcoded options", but this is a nitpick. :shrug: